src/AdvApi32.Desktop/AdvApi32.cs

```diff
@@ -888,14 +933,169 @@ public static partial class AdvApi32
+        /// <returns>If the function succeeds, the return value is <see cref="Win32ErrorCode.ERROR_SUCCESS"/>, otherwise it return one of the default error codes.</returns>
+        [DllImport(nameof(AdvApi32), CharSet = CharSet.Unicode, SetLastError = true)]
+        public static unsafe extern uint SetNamedSecurityInfo(
+            string pObjectName,
+            SE_OBJECT_TYPE ObjectType,
+            SECURITY_INFORMATION SecurityInfo,
+            IntPtr psidOwner,
+            IntPtr psidGroup,
+            [Friendly(FriendlyFlags.Out | FriendlyFlags.Optional)] Kernel32.ACL* pDacl,
+            [Friendly(FriendlyFlags.Out | FriendlyFlags.Optional)] Kernel32.ACL* pSacl);
+
+        /// <summary>
+        /// The ConvertSidToStringSid function converts a security identifier (SID) to a string format suitable for display, storage, or transmission.
+        /// To convert the string-format SID back to a valid, functional SID, call the <see cref="ConvertStringSidToSid"/> function.
+        /// </summary>
+        /// <param name="sid">A pointer to the SID structure to be converted.</param>
+        /// <param name="sidString">A pointer to a variable that receives a pointer to a null-terminated SID string. To free the returned buffer, call the <see cref="LocalFree(void*)"/> function.</param>
+        /// <returns>If the function succeeds, the return value is true, otherwise the return value is false.</returns>
+        /// <remarks>The ConvertSidToStringSid function uses the standard S-R-I-S-S… format for SID strings.</remarks>
+        [DllImport(nameof(AdvApi32), CharSet = CharSet.Unicode, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static unsafe extern bool ConvertSidToStringSid(
+            IntPtr sid,
+            [Friendly(FriendlyFlags.Array | FriendlyFlags.Out)] char* sidString);
+
+        /// <summary>
+        /// The ConvertStringSidToSid function converts a string-format security identifier (SID) into a valid, functional SID.
+        /// You can use this function to retrieve a SID that the <see cref="ConvertSidToStringSid(IntPtr, char*)"/> function converted to string format.
+        /// </summary>
+        /// <param name="StringSid">The string-format SID to convert. The SID string can use either the standard S-R-I-S-S… format for SID strings, or the SID string constant format, such as "BA" for built-in administrators.</param>
+        /// <param name="sid">A pointer to a variable that receives a pointer to the converted SID. To free the returned buffer, call the <see cref="LocalFree(void*)"/> function.</param>
+        /// <returns>If the function succeeds, the return value is true, otherwise the return value is false.</returns>
+        [DllImport(nameof(AdvApi32), CharSet = CharSet.Unicode, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool ConvertStringSidToSid(string StringSid, IntPtr sid);
```

Comment:
@vbfox , the first one is an output value and the second one is just a non-optional input value. I do agree with @AArnott  but was in doubt on how to proceed. Will change it to `string` on `ConvertStringSidToSid`